src/safeds_runner/server/pipeline_manager.py

```diff
@@ -0,0 +1,289 @@
+"""Module that contains the infrastructure for pipeline execution in child processes."""
+
+import json
+import logging
+import multiprocessing
+import queue
+import runpy
+import threading
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from multiprocessing.managers import SyncManager
+from typing import Any
+
+import simple_websocket
+import stack_data
+
+from safeds_runner.server.messages import (
+    Message,
+    MessageDataProgram,
+    create_placeholder_description,
+    create_runtime_error_description,
+    create_runtime_progress_done,
+    message_type_placeholder_type,
+    message_type_runtime_error,
+    message_type_runtime_progress,
+)
+from safeds_runner.server.module_manager import InMemoryFinder
+
+
+class PipelineManager:
+    """
+    A PipelineManager handles the execution of pipelines and processing of results.
+
+    This includes launching subprocesses and the communication between the
+    subprocess and the main process using a shared message queue.
+    """
+
+    def __init__(self) -> None:
+        """
+        Prepare the runner for running Safe-DS pipelines.
+
+        Firstly, structures shared between processes are created.
+        After that a message queue handling thread is started in the main process.
+        This allows receiving messages directly from one of the pipeline processes and relaying information
+        directly to the websocket connection (to the VS Code extension).
+        """
+        self.multiprocessing_manager: SyncManager = multiprocessing.Manager()
+        self.placeholder_map: dict = {}
+        self.messages_queue: queue.Queue[Message] = self.multiprocessing_manager.Queue()
+        self.websocket_target: simple_websocket.Server | None = None
+        self.messages_queue_thread: threading.Thread = threading.Thread(target=self._handle_queue_messages, daemon=True)
+        self.messages_queue_thread.start()
```

Comment:
Add leading `_` to the names of attributes that should be internal to the class.